### PR TITLE
Upload all unexpected tracebacks to an error channel

### LINKED ISSUE #101: Report unexpected exceptions
Add a config file setting that uploads traceback files into a debugging channel. Should also provide information on where the error occurred.

### PR DESCRIPTION
**Pull request by ammongit**
Fixes #101 

Allows the configuration of a traceback channel in the config file. If set, all unexpected errors are uploaded as a file, complete with information about the environment the command was run in. This will allow the bot to be debugged even if you weren't present for the error or it occurred in a different guild.

This option is owner-only/config-level to prevent leaking of information from other guilds.

[Example output log](https://cdn.discordapp.com/attachments/488543452305555456/504526558749392897/futaba-traceback-1540358905.log)

### CODE REVIEW

File: futaba/client.py [hunk 1]
```diff
@@ -99,6 +102,14 @@ def run_with_token(self):
         Sets up the bot's state, loads cogs then prints a 'ready' message.
         """
 
+        # Get error channel
+        if self.config.error_channel_id:
+            for channel in self.get_all_channels():
```
Comment (Ma-wa-re) [inline futaba/client.py]: Would [get_channel](https://discordpy.readthedocs.io/en/rewrite/api.html#discord.Client.get_channel) not look better here?

Comment (ammongit) [reply futaba/client.py]: Oh, good point.

**Comment by ammongit** [comment]
@Ma-wa-re since I'm aware can you give this a quick run to make sure there aren't any runtime issues?

**Comment by Ma-wa-re** [comment]
@ammongit I have ran the branch and there are no runtime issues (Check testing guild for results). As such I am merging the branch

**Pull request merged**